File: simulacoes/emulador-canais/packages/renderer-react-native-mistica/src/tokens.ts
import type { ColorValue, TextStyle, ViewStyle } from 'react-native';

export interface MobileMisticaTokens {
  colors: {
    backgroundPrimary: ColorValue;
    backgroundSecondary: ColorValue;
    surface: ColorValue;
    textPrimary: ColorValue;
    textSecondary: ColorValue;
    brand: ColorValue;
    brandPressed: ColorValue;
    onBrand: ColorValue;
    border: ColorValue;
    negative: ColorValue;
    negativeLow: ColorValue;
    positive: ColorValue;
    positiveLow: ColorValue;
    warning: ColorValue;
    warningLow: ColorValue;
    informativeLow: ColorValue;
    disabled: ColorValue;
  };
  spacing: Record<string, number>;
  iconSize: Record<string, number>;
  maxWidth: Record<string, number>;
}

export const defaultMobileMisticaTokens: MobileMisticaTokens = {
  colors: {
    backgroundPrimary: '#f4f4f6',
    backgroundSecondary: '#ececf1',
    surface: '#ffffff',
    textPrimary: '#242429',
    textSecondary: '#62626b',
    brand: '#0066cc',
    brandPressed: '#004c99',
    onBrand: '#ffffff',
    border: '#b8b8c0',
    negative: '#b42318',
    negativeLow: '#fff0f0',
    positive: '#16833b',
    positiveLow: '#e8f7ed',
    warning: '#8a5a00',
    warningLow: '#fff5d6',
    informativeLow: '#e8f2ff',
    disabled: '#d7d7dc',
  },
  spacing: {
    'spacing.none': 0,
    'spacing.xs': 4,
    'spacing.sm': 8,
    'spacing.md': 16,
    'spacing.lg': 24,
    'spacing.xl': 32,
  },
  iconSize: {
    'size.icon.sm': 16,
    'size.icon.md': 24,
    'size.control.lg': 32,
  },
  maxWidth: {
    'layout.content.compact': 480,
    'layout.content.default': 640,
    'layout.content.wide': 960,
  },
};

export function spacing(tokens: MobileMisticaTokens, token: unknown, fallback = 0): number {
  return typeof token === 'string' ? tokens.spacing[token] ?? fallback : fallback;
}

export function iconSize(tokens: MobileMisticaTokens, token: unknown): number {
  return typeof token === 'string' ? tokens.iconSize[token] ?? 24 : 24;
}

export function maxWidth(tokens: MobileMisticaTokens, token: unknown): number | undefined {
  return typeof token === 'string' ? tokens.maxWidth[token] : undefined;
}

export function color(tokens: MobileMisticaTokens, token: unknown): ColorValue | undefined {
  if (typeof token !== 'string') return undefined;
  const values: Record<string, ColorValue> = {
    'color.background.primary': tokens.colors.backgroundPrimary,
    'color.background.secondary': tokens.colors.backgroundSecondary,
    'color.text.primary': tokens.colors.textPrimary,
    'color.text.secondary': tokens.colors.textSecondary,
    'color.feedback.negative': tokens.colors.negative,
    'color.feedback.positive': tokens.colors.positive,
  };
  return values[token];
}

export function elevation(token: unknown): ViewStyle {
  if (token === 'elevation.medium') return { elevation: 6, shadowColor: '#000', shadowOpacity: 0.2, shadowRadius: 8, shadowOffset: { width: 0, height: 4 } };
  if (token === 'elevation.low') return { elevation: 2, shadowColor: '#000', shadowOpacity: 0.14, shadowRadius: 3, shadowOffset: { width: 0, height: 1 } };
  return {};
}

export function textAlign(value: unknown): TextStyle['textAlign'] {
  return value === 'center' || value === 'right' || value === 'justify' ? value : 'left';
}

File: simulacoes/emulador-canais/packages/renderer-flutter-mistica/lib/src/tokens.dart
import 'package:flutter/material.dart';

@immutable
class FlutterMisticaTokens {
  const FlutterMisticaTokens({
    this.brand = const Color(0xff0066ff),
    this.onBrand = Colors.white,
    this.background = const Color(0xfff4f4f7),
    this.backgroundSecondary = const Color(0xffe9e9ef),
    this.surface = Colors.white,
    this.textPrimary = const Color(0xff24242b),
    this.textSecondary = const Color(0xff666672),
    this.border = const Color(0xffc8c8d0),
    this.negative = const Color(0xffb3261e),
    this.negativeLow = const Color(0xffffe9e7),
    this.positive = const Color(0xff18794e),
    this.positiveLow = const Color(0xffe4f7ed),
    this.warning = const Color(0xff8a5a00),
    this.warningLow = const Color(0xfffff3cf),
    this.informativeLow = const Color(0xffe7f0ff),
  });

  final Color brand;
  final Color onBrand;
  final Color background;
  final Color backgroundSecondary;
  final Color surface;
  final Color textPrimary;
  final Color textSecondary;
  final Color border;
  final Color negative;
  final Color negativeLow;
  final Color positive;
  final Color positiveLow;
  final Color warning;
  final Color warningLow;
  final Color informativeLow;

  double spacing(dynamic token) => switch (token?.toString()) {
        'spacing.none' => 0,
        'spacing.xs' => 4,
        'spacing.sm' => 8,
        'spacing.lg' => 24,
        'spacing.xl' => 32,
        _ => 16,
      };

  double maxWidth(dynamic token) => switch (token?.toString()) {
        'layout.content.compact' => 480,
        'layout.content.wide' => 1120,
        _ => 760,
      };

  double iconSize(dynamic token) => switch (token?.toString()) {
        'size.icon.sm' => 18,
        'size.icon.lg' => 32,
        _ => 24,
      };

  Color color(dynamic token, {Color? fallback}) => switch (token?.toString()) {
        'color.background.primary' => background,
        'color.background.secondary' => backgroundSecondary,
        'color.surface' => surface,
        'color.text.primary' => textPrimary,
        'color.text.secondary' => textSecondary,
        'color.brand' || 'color.brand.primary' => brand,
        'color.feedback.negative' => negative,
        'color.feedback.positive' => positive,
        'color.feedback.warning' => warning,
        _ => fallback ?? textPrimary,
      };
}

